### PR TITLE
Add cross-release compatibility note to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ranjanan**
Add a note for package devs about porting their package from one release to another. 

cc: @tkelman @ViralBShah

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -60,6 +60,22 @@ Julia has a built-in [package manager](https://github.com/JuliaLang/METADATA.jl)
...
+on the current release, it might not work on the upcoming release or nightly.
+This is due to the fact that some Julia functions (after some discussion)
+could be deprecated or removed altogether. This may cause your package to break or
+throw a number of deprecation warnings on usage. Therefore it is highly recommended
+to port your package to latest Julia release.
+
+However, porting a package to the latest release may cause the package to break on
+earlier Julia releases. To maintain compatibility across releases, use
+[`Compat.jl`](https://github.com/JuliaLang/Compat.jl/). Find the fix for your package
+from the README, and tag the minimum version of Compat that provides the fix
```
Comment (tkelman) [inline CONTRIBUTING.md]: tag is an overloaded word here

**Comment by tkelman** [comment]
I'm not sure if this repo's CONTRIBUTING.md is the best place for info about contributing to packages.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ranjanan) [inline CONTRIBUTING.md]: Shall I use the word "specify" instead?

**Comment by ranjanan** [comment]
Maybe. @ViralBShah wanted a link to the Compat guide in CONTRIBUTING.md, so I wrote up this PR. Also, what is up with Travis? This error log says it can't find openblas. A couple of other PR's failed with seemingly unrelated errors too.

**Comment by nalimilan** [comment]
Package authors are much more likely to look at the manual section on writing packages than at CONTRIBUTING.md, which is about contributing to Julia. Why not move these explanations there?

**Comment by tkelman** [comment]
That would make sense to me.

**Comment by ranjanan** [comment]
Alright, will write up a PR for this.

**Pull request merged**